### PR TITLE
[INJICERT-1260] Added proper validation in status update endpoints

### PR DESCRIPTION
**Pull request by devshree-bhati**
### Issue Identified

- statusPurpose accepted any random/empty value and did not default to the configured value (revocation) when omitted.
- statusListIndex was not validated for being a non‑negative integer or for staying within the valid range of the status list.

### Expected Behaviour

- statusPurpose must match supported types; if missing, it should default to revocation. Invalid or empty values should return a validation error.
- statusListIndex must be a non‑negative integer and within the bounds of the corresponding status list size.

### Tests Added

- Added test cases for statusPurpose validation (valid, invalid, empty, defaulting).
- Added test cases for statusListIndex validation (negative, non‑numeric, out‑of‑range, valid cases).

## Summary by CodeRabbit

* **New Features**
  * Default status purpose applied when none is provided for credential updates.
  * Introduced an upper-bound limit for status list index checks.

* **Bug Fixes**
  * Non-negative and range validation enforced for status list references.
  * Clearer error responses via new error keys for invalid or out-of-range indices.
  * Stricter upfront validation to prevent invalid update requests.

* **Tests**
  * Expanded coverage for credential status update flows (V1/V2), including index edge cases and default-purpose behavior.

### CODE REVIEW

File: certify-core/src/main/java/io/mosip/certify/core/constants/Constants.java [hunk 1]
```diff
@@ -56,4 +56,6 @@ public class Constants {
     public static final String CREDENTIAL_OFFER_PREFIX = "credential_offer:";
     public static final String PRE_AUTHORIZED_CODE_GRANT_TYPE = "urn:ietf:params:oauth:grant-type:pre-authorized_code";
     public static final String AS_METADATA_PREFIX = "as_metadata:";
+    public static final String DEFAULT_STATUS_PURPOSE = "revocation";
```
Comment (swatigoel) [inline certify-core/src/main/java/io/mosip/certify/core/constants/Constants.java]: instead of hardcoding it in code, get it through configuration

**Review by swatigoel — CHANGES_REQUESTED** [review]

File: certify-service/src/main/java/io/mosip/certify/services/CredentialStatusServiceImpl.java [hunk 2]
```diff
@@ -39,6 +42,9 @@ public class CredentialStatusServiceImpl implements CredentialStatusService {
 
     @Override
     public CredentialStatusResponse updateCredentialStatus(UpdateCredentialStatusRequest request) {
+        String statusPurpose = validateAndGetStatusPurpose(request.getCredentialStatus() == null ? null : request.getCredentialStatus().getStatusPurpose());
+        validateStatusListIndex(request.getCredentialStatus().getStatusListIndex());
```
Comment (swatigoel) [inline certify-service/src/main/java/io/mosip/certify/services/CredentialStatusServiceImpl.java]: here also request.getCredentialStatus can be null.
we can write a block with if (request.getCredentialStatus != null) then validate otherwise set statusPurpose as null

File: certify-service/pom.xml [hunk 3]
```diff
@@ -236,8 +236,14 @@
             <artifactId>kernel-biometrics-api</artifactId>
             <version>1.3.0</version>
         </dependency>
+        <dependency>
+            <groupId>io.mosip.certify</groupId>
+            <artifactId>mock-certify-plugin</artifactId>
+            <version>0.6.0-SNAPSHOT</version>
+        </dependency>
```
Comment (swatigoel) [inline certify-service/pom.xml]: This shouldn't be committed, it's needed for local testing

File: docker-compose/docker-compose-injistack/docker-compose.yaml [hunk 4]
```diff
@@ -40,7 +40,7 @@ services:
       - ./config/certify-mock-mdl.properties:/home/mosip/config/certify-mock-mdl.properties
       - ./data/CERTIFY_PKCS12:/home/mosip/CERTIFY_PKCS12
       # This can be disabled if one uses plugins included in inji-certify-with-plugins docker image.
-#      - ./loader_path/certify/:/home/mosip/additional_jars/
+      - ./loader_path/certify/:/home/mosip/additional_jars/
```
Comment (swatigoel) [inline docker-compose/docker-compose-injistack/docker-compose.yaml]: remove this, not needed

File: db_scripts/inji_certify/deploy.properties [hunk 5]
```diff
@@ -1,4 +1,4 @@
-DB_SERVERIP=
+DB_SERVERIP=localhost
```
Comment (swatigoel) [inline db_scripts/inji_certify/deploy.properties]: don't change this

File: pom.xml [hunk 6]
```diff
@@ -3,476 +3,476 @@
    - License, v. 2.0. If a copy of the MPL was not distributed with this
    - file, You can obtain one at https://mozilla.org/MPL/2.0/. -->
 <project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-	xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
```
Comment (swatigoel) [inline pom.xml:6]: revert changes of this file as not able to identify the diff
